src/components/Register/Register.js

```diff
@@ -0,0 +1,156 @@
+  const [first, setFirst] = useState('');
+  const [last, setLast] = useState('');
+  const [email, setEmail] = useState('');
+  const [securityQuestion, setSecurityQuestion] = useState({
+    question: '',
+    answer: '',
+  });
+  const [updates, setUpdates] = useToggle(false);
+  const [specials, setSpecials] = useToggle(false);
+  const [terms, setTerms] = useToggle(false);
+
+  const getValueByType = registrationTypeFactory(
+    username,
+    password,
+    first,
+    last,
+    email,
+    specials,
+    updates,
+    securityQuestion,
+    terms,
+  );
+
+  const setStateByType = registrationTypeFactory(
+    setUsername,
+    setPassword,
+    setFirst,
+    setLast,
+    setEmail,
+    setSpecials,
+    setUpdates,
+    setSecurityQuestion,
+    setTerms,
+  );
+
+  const handleRegistrationSubmit = async () => {
+    const callbacks = data.callbacks.map(
+      ({ prompt: label, response: { input, ...all } }) => {
+        if (label === 'Select a security question') {
+          input[0].value = getValueByType(label).question;
+          input[1].value = getValueByType(label).answer;
+        } else if (!label) {
+          input[0].value = getValueByType();
+        } else {
+          input[0].value = getValueByType(label);
+        }
+        return { ...all, input };
+      },
+    );
+    try {
+      const request = {
+        callbacks,
+        authId: data.authId,
+        authServiceId: data.authServiceId,
+        header: data.pageHeader,
+        description: data.pageDescription,
+      };
+
+      const response = await ForgeRockModule.next(JSON.stringify(request));
+      if (response.type === 'LoginSuccess') {
+        navigation.navigate('Todos');
+      } else if (response.type === 'LoginFailure') {
+        // handle failure
+      }
+    } catch (err) {}
+  };
+
+  return (
+    <Box safeArea flex={1} p={2} w="90%" mx="auto">
+      <Text>Register</Text>
+      <ScrollView>
+        <FormControl isRequired>
+          {data?.callbacks.map(
+            ({
+              type,
+              prompt: label,
+              predefinedQuestions: questions,
+              response: { output },
+            }) =>
+              type === 'TermsAndConditionsCallback'
+                ? callbackTypeToComponent[type]({
+                    label: 'TermsAndConditions',
+                    terms: output[1].value,
+                    setter: setStateByType(),
+                    val: getValueByType(),
+                  })
+                : questions
+                ? callbackTypeToComponent[type]({
+                    label,
+                    questions,
+                    setter: setStateByType(label),
+                    val: getValueByType(label),
+                  })
+                : callbackTypeToComponent[type]({
+                    label,
+                    setter: setStateByType(label),
+                    val: getValueByType(label),
+                  }),
+          ) ?? null}
```

Comment:
My brain is also struggling on this. The nested(?) ternary is really tripping my brain up.